### PR TITLE
INFRA-1963 - add boto3_route53 exec and state modules

### PR DESCRIPTION
**Pull request by tkwilliams**
Adds new exec and state modules for AWS Route53 based on the newer Boto 3 API.  This allows a number of improvements and additional features by leveraging options which were not available in  Boto version 2.

Please don't merge quite yet - I'm still testing some of the code-paths which don't generally get exercised in our environment, and mostly just wanna get some eyeballs on the code and linting from upstream :)  Will prolly be ready to merge in a day or two.

### CODE REVIEW

**Comment by cachedout** [comment]
@tkwilliams OK. Please keep me posted.

**Comment by tkwilliams** [comment]
This is stable and working, BUT I'm not happy with how it's parsing / unparsing non-alphanum chars in RRs (that is to say, it's really not - you have to pass e.g. an asterisk as EXACTLY \\\052 else the yaml parser mis-parses it in various interesting ways.  E.g. 
```
>>> import yaml
>>> yaml.load('{foo: \\052.foo.org.}')
{'foo': '\\052.foo.org.'}
>>> yaml.load('{foo: "\\052.foo.org."}')
{'foo': '\x0052.foo.org.'}
>>> yaml.load('{foo: !!str \\052.foo.org.}')
{'foo': '\\052.foo.org.'}
>>> yaml.load('{foo: "\\052.foo.org."}')
{'foo': '\x0052.foo.org.'}
>>> yaml.load('{foo: "*.foo.org."}')
{'foo': '*.foo.org.'}
>>> yaml.load('{foo: "\052.foo.org."}')
{'foo': '*.foo.org.'}
>>> yaml.load('{foo: \052.foo.org.}')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/site-packages/yaml/__init__.py", line 71, in load
    return loader.get_single_data()
  File "/usr/lib64/python2.7/site-packages/yaml/constructor.py", line 37, in get_single_data
    node = self.get_single_node()
  File "/usr/lib64/python2.7/site-packages/yaml/composer.py", line 35, in get_single_node
    if not self.check_event(StreamEndEvent):
  File "/usr/lib64/python2.7/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/usr/lib64/python2.7/site-packages/yaml/parser.py", line 143, in parse_implicit_document_start
    StreamEndToken):
  File "/usr/lib64/python2.7/site-packages/yaml/scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "/usr/lib64/python2.7/site-packages/yaml/scanner.py", line 224, in fetch_more_tokens
    return self.fetch_alias()
  File "/usr/lib64/python2.7/site-packages/yaml/scanner.py", line 611, in fetch_alias
    self.tokens.append(self.scan_anchor(AliasToken))
  File "/usr/lib64/python2.7/site-packages/yaml/scanner.py", line 930, in scan_anchor
    % ch.encode('utf-8'), self.get_mark())
yaml.scanner.ScannerError: while scanning an alias
  in "<string>", line 1, column 7:
    {foo: *.foo.org.}
          ^
expected alphabetic or numeric character, but found '.'
  in "<string>", line 1, column 8:
    {foo: *.foo.org.}
           ^
>>>
```
Well, actually the cases where it parses it into "*.foo.org." are valid, but due to the issue mentioned next it currently doesn't work, so again I need to polish this before merging...

BTW, the one where it turns into 4-digit HEX is particularly interesting.  I can't find any mention of such behaviour on yaml.org so I'd call this one a definite boog.

Next, AWS returns errors for some things which it's docu says it should accept - I'll need to spend some time reverse-engineering its behaviour and update the _to_aws_encoding() func to deal with it.  Lastly I need to actually implement the _from_aws_encoding() func to permit idempotent comparisons, otherwise we'll always be forced to pass some otherwise acceptable characters as octal escapes.

TL;DR - I'll need a couple more days on this before it's fit to merge... :)

**Comment by cachedout** [comment]
@tkwilliams Sounds good. Thank you very much for the update!

**Comment by tkwilliams** [comment]
Grrr.  Because this is in a "works well enough for us" state, I've gotten sucked into other tasks the past few days.  I REALLY want to get back and finish this up in the next couple of days (or perhaps evenings would be more honest).  If not, @cachedout would you rather go ahead and merge it (it DOES work, just not optimally in the special case of "funky" characters in record names) or wait a bit longer until I can make it perfect?  :)

**Comment by cachedout** [comment]
I think that the right thing to do here is to get this in and then iterate on it. It's in the develop branch, of course, and that's the place for stuff like this. Thanks very much for this, @tkwilliams . I'm looking forward to having this in and see the upcoming improvements as well.
